### ISSUE TITLE
Create a Transactions view

### ISSUE THREAD

**Issue by 37Rb** [37Rb/nextcloud-hledger#15]
Create a view that lists all transactions with search and filter capabilities.

**Comment by simonmichael** [comment]
This and some of the other issues sound like they are rebuilding hledger-web. That may be a good thing to attempt - because it doesn't fit nextcloud's needs or simply to build a better ui - I just wanted to make sure you know of it.

**Comment by thecount2a** [comment]
Since my workflow requires editing transactions, I've actually created a fully-featured transaction editor including the ability to edit very large ledger files efficiently and without corrupting ledger files that contain unknown directives. It uses a Vue UI toolkit called Vuetify though so if we end up merging this, I will probably follow it up soon by making the entire nextcloud UI use Vuetify too so that the UI is consistent across the board. Here is a screenshot of my current progress on this! Bonus, it's 100% mobile friendly and reactive to large and small screens and also tablets and the smaller cell phone screens.

Even though I've opened a pull request, I'm not quite ready to merge it. I still need to solve one more conflict between Nextcloud UI toolkit CSS and the Vuetify CSS. I'm putting it up here so that I can get any comments on changes that people might want to see and comments on introducing Vuetify js/CSS libraries. Is this a good idea? It's definitely prettier than being limited to the UI offered through Nextcloud. Hopefully it won't jeopardize our chances of being admitted to the Nextcloud app store.

Pull request so anybody can see a diff: https://github.com/37Rb/nextcloud-hledger/pull/21
Working branch: https://github.com/37Rb/nextcloud-hledger/tree/new-editor
Vuetify website, for reference: https://vuetifyjs.com/en/

![2021-08-25-172403_1473x929_scrot](https://user-images.githubusercontent.com/15679291/130866669-35b5d30a-5ffd-4d7f-876f-9d57a6064ba8.png)

**Comment by simonmichael** [comment]
Very cool @thecount2a. This works only for nextcloud users right now, I guess ? By the way, is hledger available to nextcloud users yet, or still in preparation ?

**Comment by thecount2a** [comment]
@simonmichael yes, this editor is currently only for nextcloud users although I originally wrote it as a standalone Vue (javascript) app which I pushed to github here: https://github.com/thecount2a/ledgereditor The only caveat is that, while porting the ledger editor into the Nextcloud environment, I made a bunch of improvements which I have not yet backported into the original straight javascript repository (thecount2a/ledgereditor) yet. Once I've backported those changes to the original repo, perhaps I will host the ledger editor on Github pages for anybody to use who wants to use a structured GUI editor to edit ledger files without uploading them to any external server. It works by allowing the user to copy and paste a ledger file into a text area, then it builds the GUI editor to allow real-time editing the ledger. The user can go back to the raw text-based ledger text box and copy&paste the newly edited ledger text back out again to "export" the final ledger copy to store on the local filesystem again. It's a bit cumbersome to use for everyday editing which is why it is best paired with nextcloud where it can save the modified ledger back into the nextcloud file storage. That being said, I'm trying to make it modular enough so that multiple web-based ledger projects could import and use the editor if they wanted to. hledger-web could probably offer this editor too, if you wanted to but the GUI wouldn't look very consistent unless you switched the entire hledger-web site to use Vue/Vuetify which would be a big re-write.

Regarding submitting the hledger app to the Nextcloud app store, we have a milestone setup to track our progress toward being in a place where we feel "ready" to begin the process of submitting it: https://github.com/37Rb/nextcloud-hledger/milestone/1

Until those items are finished (specifically the security audit task), we might get rejected by the Nextcloud app store anyway so we want to make sure everything is ready before trying that. Short answer is, it's still in preparation. Any JS/Vue developers who want to help out, feel free to take a look at that milestone. It's going slow since @37Rb has a new job and I have a new baby at home so neither of us are spending as much time as we thought we would be able to. Still, Ryan told me he has been using this nextcloud app for his personal finances so he certainly hasn't abandoned it. I am VERY close being able to use it for my personal finances too but I need to finalize my transaction import workflow before I can do that. I will begin using it soon. As soon as I begin using it myself, I'm sure that will accelerate the development a bit too, so I can work out some of the kinks. 

For anybody wanting to try out the current development version of this app, setting up a nextcloud server on a local docker instance and checking out and building the development version of this app is not hard. Maybe I will document how to do that from scratch in the README when I get a few spare moments.

**Comment by simonmichael** [comment]
Awesome. No pressure on this, thanks for your work and the update (and congrats!!)

**Comment by 37Rb** [comment]
WOW!!! Nice work @thecount2a ! I can't wait to try it! I've had a few things hogging my attention recently. Started a new job and also moved to a new state. Yes, I want to get back into developing this.
